service/src/main/java/io/camunda/service/ElementInstanceServices.java

```diff
@@ -62,12 +71,12 @@ public SearchQueryResult<FlowNodeInstanceEntity> search(
   }
 
   public FlowNodeInstanceEntity getByKey(final Long key) {
+    final var query = FlowNodeInstanceQuery.of(q -> q.filter(f -> f.flowNodeInstanceKeys(key)));
     final var result =
-        flowNodeInstanceSearchClient
-            .withSecurityContext(securityContextProvider.provideSecurityContext(authentication))
-            .searchFlowNodeInstances(
-                flownodeInstanceSearchQuery(q -> q.filter(f -> f.flowNodeInstanceKeys(key))));
+        search(query, securityContextProvider.provideSecurityContext(authentication));
+
     final var flowNodeInstance = getSingleResultOrThrow(result, key, "Flow node instance");
+
     final var authorization = Authorization.of(a -> a.processDefinition().readProcessInstance());
     if (!securityContextProvider.isAuthorized(
         flowNodeInstance.processDefinitionId(), authentication, authorization)) {
```

Comment:
🔧 I believe we can remove this auth check and generally add it in the private `search` method. If the user has no permission, no result will be returned by the data layer. We will run into NOT_FOUND then in `getSingleResultOrThrow`. This is actually better than returning "not authorized".

Reply:
Yeah, that has been also my thoughts, but for now I just tried to keep the old behavior. Maybe we should make a switch from `forbidden` to `not_found` in a separate sweep all PR. wdyt?

Reply:
Yeah, that makes sense. Let's make it a dedicated effort that's documented and aligned on 👍